### PR TITLE
Remove all 'Std' from the transport project

### PR DESCRIPTION
**Pull request by kxu1026**

### CODE REVIEW

File: docs/authoring-transport-udfs.md [hunk 1]
```diff
@@ -56,8 +55,8 @@ public class Multiply extends StdUDF2<StdInteger, StdInteger, StdInteger>
   }
 
   @Override
-  public StdInteger eval(StdInteger first, StdInteger second) {
-    return getStdFactory().createInteger(first.get() * second.get());
+  public Intger eval(Intger first, Intger second) {
+    return first.get() * second.get();
```
[the remark this answers is not in the document]
Comment (kxu1026) [reply docs/authoring-transport-udfs.md]: addressed.

File: docs/authoring-transport-udfs.md [hunk 2]
```diff
@@ -72,10 +71,8 @@ public class Multiply extends StdUDF2<StdInteger, StdInteger, StdInteger>
 }
 ```
 
-In the example above, `StdInteger` is an interface that provides high-level integer operations to its objects.
 Depending on the engine where this UDF is executed, this interface is implemented differently to deal with native data types used by that engine.
-`getStdFactory()` is a method used to create objects that conform to a given data type.
```
Comment (wmoustafa) [inline docs/authoring-transport-udfs.md:77]: This got dropped

File: docs/required-trino-apis.md [hunk 3]
```diff
@@ -3,12 +3,12 @@ Transport requires applying this [patch](transport-udf-trino.patch) before being
 This patch makes some of the internal UDF classes be visible at the SPI layer. 
 Below we explain why some Transport APIs cannot leverage the APIs offered by the [public SPI UDF model](https://trino.io/docs/current/develop/functions.html).
 
-## [init() method](https://github.com/linkedin/transport/blob/09a89508296a2491f43cc8866d47952c911313ab/transportable-udfs-api/src/main/java/com/linkedin/transport/api/udf/StdUDF.java#L45) is hard to implement on top of Trino-SPI
+## [init() method](https://github.com/linkedin/transport/blob/09a89508296a2491f43cc8866d47952c911313ab/transportable-udfs-api/src/main/java/com/linkedin/transport/api/udf/UDF.java#L45) is hard to implement on top of Trino-SPI
```
Comment (wmoustafa) [inline docs/required-trino-apis.md:6]: Could you make sure all the links are still valid?

File: docs/writing-tests.md [hunk 4]
```diff
@@ -59,7 +59,7 @@ Running `gradle build` from the terminal will build the UDF artifacts as well as
 
 ## More Complex Tests
 
-To figure out Java equivalents of StdData types to be used in the test framework, please see the documentation of [AbstractStdUDFTest](../transportable-udfs-test/transportable-udfs-test-api/src/main/java/com/linkedin/transport/test/AbstractStdUDFTest.java).
+To figure out Java equivalents of Data types to be used in the test framework, please see the documentation of [AbstractUDFTest](../transportable-udfs-test/transportable-udfs-test-api/src/main/java/com/linkedin/transport/test/AbstractStdUDFTest.java).
```
Comment (wmoustafa) [inline docs/writing-tests.md:62]: small `data`

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala [hunk 5]
```diff
@@ -17,44 +17,44 @@ import scala.collection.JavaConverters._
 
 class TestSparkTypeFactory {
 
-  val typeFactory: typesystem.SparkTypeFactory = new typesystem.SparkTypeFactory
-  val stdFactory = new SparkTypeFactory(new SparkBoundVariables)
+  val typeSystemTypeFactory: typesystem.SparkTypeFactory = new typesystem.SparkTypeFactory
```
Comment (wmoustafa) [inline transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala:20]: `typeSystemTypeFactory` --> `abstractSparkTypeFactory`

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala [hunk 6]
```diff
@@ -64,23 +64,23 @@ class TestSparkTypeFactory {
       "bytesField", "arrField")
     val fieldTypes = Array("varchar", "integer", "bigint", "boolean", "real", "double", "varbinary", "array(integer)")
 
-    val stdStruct = stdFactory.createStruct(fieldNames.toList.asJava, fieldTypes.map(stdFactory.createDataType).toList.asJava)
-    val internalRow = stdStruct.asInstanceOf[PlatformData].getUnderlyingData.asInstanceOf[InternalRow]
+    val row = typeFactory.createStruct(fieldNames.toList.asJava, fieldTypes.map(typeFactory.createDataType).toList.asJava)
```
Comment (wmoustafa) [inline transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala:67]: `createStruct` --> `createRowData`

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkArray.scala [hunk 7]
```diff
@@ -15,41 +15,41 @@ import org.testng.annotations.Test
...
+    assertEquals(row.size(), arrayData.numElements() + 1)
+    assertEquals(row.get(row.size() - 1), 5)
   }
 
   @Test
   def testSparkArrayMutabilityReset(): Unit = {
-    val stdArray = SparkConverters.toTransportData(arrayData, arrayType).asInstanceOf[data.ArrayData[Integer]]
-    stdArray.add(5)
-    stdArray.asInstanceOf[PlatformData].setUnderlyingData(arrayData)
+    val row = SparkConverters.toTransportData(arrayData, arrayType).asInstanceOf[data.ArrayData[Integer]]
```
Comment (wmoustafa) [inline transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkArray.scala:49]: Here and in other places you are renaming an array to a row.

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkRowData.scala [hunk 8]
```diff
@@ -25,43 +25,43 @@ class TestSparkRowData {
 
   @Test
   def testCreateSparkStruct(): Unit = {
-    val stdStruct = SparkConverters.toTransportData(structData, structType).asInstanceOf[RowData]
-    assertSame(stdStruct.asInstanceOf[PlatformData].getUnderlyingData, structData)
+    val rowData = SparkConverters.toTransportData(structData, structType).asInstanceOf[RowData]
```
Comment (wmoustafa) [inline transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkRowData.scala:28]: There are other places with `row` as opposed to `rowData`. Let us use `rowData`.

File: docs/authoring-transport-udfs.md (hunk 2 above)
Comment (kxu1026) [reply docs/authoring-transport-udfs.md:77]: addressed

File: docs/required-trino-apis.md (hunk 3 above)
Comment (kxu1026) [reply docs/required-trino-apis.md:6]: Why do we verify that right now? As I mentioned, we should verify after we change the branch. I am not comfortable if the link will still be good or not after renaming the branch.

File: docs/writing-tests.md (hunk 4 above)
Comment (kxu1026) [reply docs/writing-tests.md:62]: addressing

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala (hunk 5 above)
Comment (kxu1026) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala:20]: addressing

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala (hunk 6 above)
Comment (kxu1026) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala:67]: That's from the TypeFactory Interface. I am assuming you want to rename 'createStruct' to 'createRowData' for everything.

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkArray.scala (hunk 7 above)
Comment (kxu1026) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkArray.scala:49]: addressing

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkRowData.scala (hunk 8 above)
Comment (kxu1026) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkRowData.scala:28]: ok

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala (hunk 5 above)
Comment (kxu1026) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala]: addressed

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala (hunk 6 above)
Comment (kxu1026) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala]: addressed

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkArray.scala (hunk 7 above)
Comment (kxu1026) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkArray.scala]: addressed

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkRowData.scala (hunk 8 above)
Comment (kxu1026) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkRowData.scala:28]: addressed

File: docs/authoring-transport-udfs.md [hunk 9]
```diff
@@ -91,8 +89,8 @@ For instructions on how to use these artifacts, see [Using Transport UDFs](using
     - Transport UDFs can accept/return complex types. E.g. the input parameter signature for a UDF which accepts a list of integers would be `array(integer)`. You can also use generic types to derive types at runtime. E.g. you can accept a generic type `K` as input and return an `array(K)` in which case the type of `K` will be derived at query compile time.
     - Example: [MapFromTwoArraysFunction](../transportable-udfs-examples/transportable-udfs-example-udfs/src/main/java/com/linkedin/transport/examples/MapFromTwoArraysFunction.java) and [StructCreateByNameFunction](../transportable-udfs-examples/transportable-udfs-example-udfs/src/main/java/com/linkedin/transport/examples/StructCreateByNameFunction.java)
 - UDF overloading
-    - You can define multiple Transport UDFs which share the same name but accept different input parameter signatures using the [`TopLevelStdUDF` Interface](transport-udfs-api.md#toplevelstdudf-interface).
+    - You can define multiple Transport UDFs which share the same name but accept different input parameter signatures using the [`TopLevelUDF` Interface](transport-udfs-api.md#topleveludf-interface).
```
Comment (wmoustafa) [inline docs/authoring-transport-udfs.md:92]: Does this link need to be fixed? I do not see the link itself changing. Please address similar issues.

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala [hunk 10]
```diff
@@ -17,44 +17,44 @@ import scala.collection.JavaConverters._
...
   @Test
-  def testCreateStructFromStdType(): Unit = {
+  def testCreateStructFromType(): Unit = {
     val fieldNames = Array("strField", "intField", "longField", "boolField", "floatField", "doubleField",
       "bytesField", "arrField")
     val fieldTypes = Array("varchar", "integer", "bigint", "boolean", "real", "double", "varbinary", "array(integer)")
 
-    val stdStruct = stdFactory.createStruct(stdFactory.createDataType(fieldNames.zip(fieldTypes).map(x => x._1 + " " + x._2).mkString("row(", ", ", ")")))
-    val internalRow = stdStruct.asInstanceOf[PlatformData].getUnderlyingData.asInstanceOf[InternalRow]
+    val rowData = typeFactory.createRowData(typeFactory.createDataType(fieldNames.zip(fieldTypes).map(x => x._1 + " " + x._2).mkString("row(", ", ", ")")))
```
Comment (wmoustafa) [inline transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala:53]: We should be consistent (regarding whether to use `Data` suffix or not)
* map, array, rowData should all use the same convention
* createArray, createMap, createRowData should all use the same convention
Let us discuss offline.

File: docs/authoring-transport-udfs.md (hunk 9 above)
Comment (kxu1026) [reply docs/authoring-transport-udfs.md:92]: This link has been fixed. what's the issue?

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala (hunk 10 above)
Comment (kxu1026) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala:53]: can we do a quick sync?

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkRowData.scala (hunk 8 above)
Comment (wmoustafa) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkRowData.scala:28]: Could you update `array` and `map` to `arrayData` and `mapData`?

**Review by wmoustafa — CHANGES_REQUESTED** [review]

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkRowData.scala (hunk 8 above)
Comment (kxu1026) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkRowData.scala:28]: please more specific for your request. I don't think there is any array and map in this TestSparkRowData class. And, arrayData and mapData are applied properly in the project.

Comment (wmoustafa) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkRowData.scala:28]: They are not applied properly in the project. Please take a look at other classes in this patch.

**Review by wmoustafa — CHANGES_REQUESTED** [review]

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala (hunk 10 above)
[the remark this answers is not in the document]
Comment (kxu1026) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala]: addressed.

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala [hunk 11]
```diff
@@ -17,44 +17,44 @@ import scala.collection.JavaConverters._
...
+    val internalRow = rowData.asInstanceOf[PlatformData].getUnderlyingData.asInstanceOf[InternalRow]
```
[the remark this answers is not in the document]
Comment (kxu1026) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala]: addressed.

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala [hunk 12]
```diff
@@ -64,23 +64,23 @@ class TestSparkTypeFactory {
       "bytesField", "arrField")
     val fieldTypes = Array("varchar", "integer", "bigint", "boolean", "real", "double", "varbinary", "array(integer)")
 
-    val stdStruct = stdFactory.createStruct(fieldNames.toList.asJava, fieldTypes.map(stdFactory.createDataType).toList.asJava)
-    val internalRow = stdStruct.asInstanceOf[PlatformData].getUnderlyingData.asInstanceOf[InternalRow]
+    val rowData = typeFactory.createRowData(fieldNames.toList.asJava, fieldTypes.map(typeFactory.createDataType).toList.asJava)
```
[the remark this answers is not in the document]
Comment (kxu1026) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala]: addressed.

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala [hunk 13]
```diff
@@ -64,23 +64,23 @@ class TestSparkTypeFactory {
...
     })
   }
 
   @Test
   def testCreateStructFromFieldTypes(): Unit = {
     val fieldTypes = Array("varchar", "integer", "bigint", "boolean", "real", "double", "varbinary ", "array(integer)")
 
-    val stdStruct = stdFactory.createStruct(fieldTypes.map(stdFactory.createDataType).toList.asJava)
-    val internalRow = stdStruct.asInstanceOf[PlatformData].getUnderlyingData.asInstanceOf[InternalRow]
+    val rowData = typeFactory.createRowData(fieldTypes.map(typeFactory.createDataType).toList.asJava)
```
[the remark this answers is not in the document]
Comment (kxu1026) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala]: addressed.

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala [hunk 14]
```diff
@@ -64,23 +64,23 @@ class TestSparkTypeFactory {
...
+    val internalRow = rowData.asInstanceOf[PlatformData].getUnderlyingData.asInstanceOf[InternalRow]
```
[the remark this answers is not in the document]
Comment (kxu1026) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/TestSparkTypeFactory.scala]: addressed.

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkRowData.scala (hunk 8 above)
[the remark this answers is not in the document]
Comment (kxu1026) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkRowData.scala]: addressed.

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkRowData.scala [hunk 15]
```diff
@@ -25,43 +25,43 @@ class TestSparkRowData {
...
   }
 
   @Test
   def testSparkStructSetField(): Unit = {
-    val stdStruct = SparkConverters.toTransportData(structData, structType).asInstanceOf[RowData]
-    stdStruct.setField(1, 1)
-    assertEquals(stdStruct.getField(1), 1)
-    stdStruct.setField(fieldNames(2), 5L) // scalastyle:ignore magic.number
-    assertEquals(stdStruct.getField(fieldNames(2)), 5L) // scalastyle:ignore magic.number
+    val rowData = SparkConverters.toTransportData(structData, structType).asInstanceOf[RowData]
```
[the remark this answers is not in the document]
Comment (kxu1026) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkRowData.scala]: addressed.

File: transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkRowData.scala [hunk 16]
```diff
@@ -25,43 +25,43 @@ class TestSparkRowData {
...
-    assertNotSame(stdStruct.asInstanceOf[PlatformData].getUnderlyingData, structData)
+    assertNotSame(rowData.asInstanceOf[PlatformData].getUnderlyingData, structData)
   }
 
   @Test
   def testSparkStructMutabilityReset(): Unit = {
-    val stdStruct = SparkConverters.toTransportData(structData, structType).asInstanceOf[RowData]
-    stdStruct.setField(1, 1)
-    stdStruct.asInstanceOf[PlatformData].setUnderlyingData(structData)
+    val rowData = SparkConverters.toTransportData(structData, structType).asInstanceOf[RowData]
```
[the remark this answers is not in the document]
Comment (kxu1026) [reply transportable-udfs-spark_2.11/src/test/scala/com/linkedin/transport/spark/data/TestSparkRowData.scala]: addressed.

**Review by wmoustafa — APPROVED** [review]

**Pull request merged**